### PR TITLE
feat (xquery): implement fn:parse-ietf-date()

### LINKED ISSUE #2820: implement fn:parse-ietf-date
### What is the problem

`fn:parse-ietf-date($value as xs:string?) as xs:dateTime?` is not defined in xpath-functions module.
This function is a helper to deal with date time strings received in HTTP headers like `Date`, `Last-Modified` and `If-modified-since' etc.

I was able to implement this partly in xquery. The code is in this [gist](https://gist.github.com/line-o/13c8224f7fa8938f80b6a640da134eeb)
Note that `imfd:to-dateTime` is limited to handle strings in the fix-date subset of  [Internet Message Format](https://tools.ietf.org/html/rfc5322).

### What did you expect

The function to be implemented.

BaseX has implemented it:

`XQUERY fn:parse-ietf-date('Tue, 18 Jun 2019 13:10:56 GMT')`
returns `2019-06-18T13:10:56Z`

spec: https://www.w3.org/XML/Group/qtspecs/specifications/xpath-functions-31/html/Overview.html#func-parse-ietf-date

### Describe how to reproduce or add a test

Eval `fn:parse-ietf-date('Tue, 18 Jun 2019 13:10:56 GMT')` in eXide

### Context information
Please always add the following information
- eXist-db 5.0.0-RC8 - commit 0b53dd1ba552cc329e0737dd6869ceb348ac1a4b 
- Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
- Operating system MacOs
- 64 bit
- clone from GitHub
-  no custom changes in e.g. conf.xml

### PR DESCRIPTION
**Pull request by line-o**
# Description:

Fully [spec](https://www.w3.org/XML/Group/qtspecs/specifications/xpath-functions-31/html/Overview.html#func-parse-ietf-date) compliant implementation of `fn:parse-ietf-date()` in java.

- add error code FORG0010
- add function to fn namespace

### Reference:

closes #2820

### Type of tests:

Xquery testsuite

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunParseIetfDate.java [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ *  $Id$
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/functions/fn/FunParseIetfDate.java]: can be left out

**Comment by dizzzz** [comment]
yes, this is a nice and robust solution.

**Comment by duncdrum** [comment]
@line-o can you remove the function from the list of unsupported features in the documenation pls. [here](https://github.com/eXist-db/documentation/blob/1b81fb207669d4b89e95f94e4ce479d77267908e/src/main/xar-resources/data/xquery/xquery.xml#L146)

**Comment by line-o** [comment]
@duncdrum It is a little unclear where exactly I should remove that function from the list of unsupported ones. (master, 1b81fb, 5.0.0-RC1, ...)

**Comment by line-o** [comment]
That last issue Codacy is complaining about, complexity in tzoffset(), is a little picky in my opinion.
Any proposals how to make that service happy?

**Comment by line-o** [comment]
Also, could someone re-run the travis tests? This time Java8 and Java9 builds failed for no apparent reason.

**Comment by line-o** [comment]
*sigh*, I was able to simplify `tzoffset()` but IDEA warned me to make the FunctionSignature package-private which Codacy is now complaining about. Any advice which one of them is right?

**Comment by line-o** [comment]
Also, I ran some speed tests and it turned out that `ZonedDateTime.parse()` was actually a tad slower than the custom parser (~1 microsecond).
I removed it therefore, which further simplifies the implementation.

**Pull request merged**